### ISSUE TITLE
systemd-vconsole-setup.service seems to get started after every switch

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nixpkgs#23118]
I just built an EC2 image from a recent master and booted it up. It seems to tell me a "new unit" is started after every switch. I'd expect it to call it a new unit once and then not start it every time.

Version:

```
# nixos-version
17.03.git.15c05ad (Gorilla)
```

My configuration:

```nix
{ pkgs, lib, ... }:

{
  imports = [ <nixpkgs/nixos/modules/virtualisation/amazon-image.nix> ];
  ec2.hvm = true;

  services.httpd = {
    enable = true;
    adminAddr = "test@example.org";
    documentRoot = "${pkgs.valgrind.doc}/share/doc/valgrind/html";
  };
  networking.firewall.allowedTCPPorts = [ 80 ];
}
```

```
[root@host:~]# nixos-rebuild switch
building Nix...
building the system configuration...
updating GRUB 2 menu...
activating the configuration...
setting up /etc...
setting up tmpfiles
the following new units were started: httpd.service, keys.target, systemd-vconsole-setup.service

[root@host:~]# nixos-rebuild switch
building Nix...
building the system configuration...
updating GRUB 2 menu...
activating the configuration...
setting up /etc...
setting up tmpfiles
the following new units were started: systemd-vconsole-setup.service

[root@host:~]# nixos-rebuild switch
building Nix...
building the system configuration...
updating GRUB 2 menu...
activating the configuration...
setting up /etc...
setting up tmpfiles
the following new units were started: systemd-vconsole-setup.service
```

**Comment by abbradar** [comment]
I'm not sure but is there any graphical terminal on EC2? Then we need to disable `setVconsole` from `kbd.nix` but there isn't an option for that currently...

**Comment by copumpkin** [comment]
No graphical terminal on EC2. There's an API to capture screenshots of what the screen would look like if it did have a screen 😄

Was this `vconsole` thing added recently? I've been running EC2 NixOS images for a while and I don't think I've seen this before.

**Comment by rnhmjoj** [comment]
I noticed this as well. maybe it was introduce here: https://github.com/NixOS/nixpkgs/commit/6c11d9dd55b7dab91fe54ab17f739df74f9b4b2c

**Comment by abbradar** [comment]
@copumpkin Nope, it was there for all time. I'm actually not sure why did it start to behave like this now -- the only recent change was setting `/dev/tty1` in 0c81594a29d99d9ee9c9c88c680340d56823cfb2 which shouldn't make it worse. https://github.com/NixOS/nixpkgs/commit/6c11d9dd55b7dab91fe54ab17f739df74f9b4b2c shouldn't be an issue too -- it essentially only adds an additional restart trigger.

**Comment by copumpkin** [comment]
Interesting. I'll take a look at older AMIs; could be I just tuned out the noise 😄 

Either way it would be nice to make it go away. Do other distros enable this by default on EC2?

**Comment by abbradar** [comment]
@copumpkin A simple workaround would be to disable it but I'm not sure why this happens in first place. Also is there is an API to make screenshots there is _some_ virtual terminal, so technically `systemd-vconsole-setup` should be fine...

Can you check `systemctl status systemd-vconsole-setup`?

**Comment by copumpkin** [comment]
```
$ systemctl status systemd-vconsole-setup
● systemd-vconsole-setup.service - Setup Virtual Console
   Loaded: loaded (/nix/store/lm7y21wfj79bh0vcyd8rc9bf2lrpcbsf-systemd-232/example/systemd/system/systemd-vconsole-setup.service; enabled; vendor preset: enabled)
  Drop-In: /nix/store/8bcslnycas9aqi8avy2crca8z83pirnj-system-units/systemd-vconsole-setup.service.d
           └─overrides.conf
   Active: inactive (dead) since Thu 2017-02-23 20:26:22 UTC; 19h ago
     Docs: man:systemd-vconsole-setup.service(8)
           man:vconsole.conf(5)
  Process: 5008 ExecStart=/nix/store/lm7y21wfj79bh0vcyd8rc9bf2lrpcbsf-systemd-232/lib/systemd/systemd-vconsole-setup /dev/tty1 (code=exited, status=0/SUCCESS)
 Main PID: 5008 (code=exited, status=0/SUCCESS)

Feb 23 20:26:22 hostname systemd[1]: Starting Setup Virtual Console...
Feb 23 20:26:22 hostname systemd[1]: Started Setup Virtual Console.
```

```
$ journalctl -u systemd-vconsole-setup
-- Logs begin at Thu 2017-02-23 19:51:54 UTC, end at Thu 2017-02-23 20:26:22 UTC. --
Feb 23 19:51:59 hostname systemd[1]: Started Setup Virtual Console.
Feb 23 19:53:36 hostname systemd[1]: Starting Setup Virtual Console...
Feb 23 19:53:36 hostname systemd[1]: Started Setup Virtual Console.
Feb 23 20:17:37 hostname systemd[1]: Starting Setup Virtual Console...
Feb 23 20:17:37 hostname systemd[1]: Started Setup Virtual Console.
Feb 23 20:26:05 hostname systemd[1]: Starting Setup Virtual Console...
Feb 23 20:26:06 hostname systemd[1]: Started Setup Virtual Console.
Feb 23 20:26:22 hostname systemd[1]: Starting Setup Virtual Console...
Feb 23 20:26:22 hostname systemd[1]: Started Setup Virtual Console.
```

**Comment by abbradar** [comment]
Hm, not sure why it's started each time then -- it's a oneshot unit that surely hasn't changed between switches. May be a bug in `switch-to-configuration.pl`.

**Comment by copumpkin** [comment]
I'll try to debug a bit later 😄

**Comment by copumpkin** [comment]
@abbradar when you run this outside of EC2, does that service show up as active or inactive? Trying to find diffs in our state.

**Comment by abbradar** [comment]
Inactive -- it shouldn't be active as it's a oneshot service without `RemainOnExit = true`. BTW it's double strange that I can't reproduuce that on a physical machine. FWIW I'm on 183eeb3c0fdac8de3146aedaa6028b474f96db6f.

**Comment by pbogdan** [comment]
FWIW I can reproduce this on my laptop running 17.03. Oddly enough if I execute `nixos-rebuild switch` from tty1 it prints `the following new units were started: systemd-vconsole-setup.service` each time. If I execute it from any other tty or from graphical session I can't get the message to show at all even though ` systemctl status systemd-vconsole-setup.service` reports the service has been started again. Not sure what to make of that but thought I would leave a comment in case it provides any clues.

**Comment by pbogdan** [comment]
This seems to have something to do with `sysinit.target` being started (reached?) during `nixos-rebuild switch`, here's what I'm seeing with `dbus-monitor --system`:

```
signal time=1492539496.521781 sender=:1.1 -> destination=(null destination) serial=18151 path=/org/freedesktop/systemd1; interface=org.freedesktop.systemd1.Manager; member=JobNew
   uint32 9617
   object path "/org/freedesktop/systemd1/job/9617"
   string "sysinit.target"

...

signal time=1492539496.521803 sender=:1.1 -> destination=(null destination) serial=18155 path=/org/freedesktop/systemd1; interface=org.freedesktop.systemd1.Manager; member=JobNew
   uint32 9648
   object path "/org/freedesktop/systemd1/job/9648"
   string "systemd-vconsole-setup.service"
```

and `systemd-vconsole-setup.service` being its dependency:

<pre>
systemctl list-dependencies --no-legend sysinit.target 
sysinit.target
● ├─dev-hugepages.mount
● ├─dev-mqueue.mount
● ├─firewall.service
● ├─kmod-static-nodes.service
<b>● ├─sys-fs-fuse-connections.mount</b>
<b>● ├─sys-kernel-config.mount</b>
● ├─sys-kernel-debug.mount
● ├─systemd-ask-password-console.path
<b>● ├─systemd-binfmt.service</b>
● ├─systemd-journal-catalog-update.service
● ├─systemd-journal-flush.service
● ├─systemd-journald.service
● ├─systemd-modules-load.service
● ├─systemd-random-seed.service
● ├─systemd-sysctl.service
● ├─systemd-timesyncd.service
● ├─systemd-tmpfiles-setup.service
● ├─systemd-udev-trigger.service
● ├─systemd-udevd.service
● ├─systemd-update-utmp.service
<b>● └─systemd-vconsole-setup.service</b>
</pre>

All of the entries highlighted in bold are in `inactive (dead)` state and it looks like systemd is trying to start them all.

**Comment by globin** [comment]
Fixed, see #22470.
